scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,310 @@
+import heapq
+import itertools
+
+from dataclasses import dataclass
+
+import numpy as np
+
+from scipy.integrate._rules import (
+    ProductNestedFixed, NestedFixedRule,
+    GaussKronrodQuad, NewtonCotesQuad,
+)
+
+
+@dataclass
+class CubatureRegion:
+    estimate: np.ndarray
+    error: np.ndarray
+    a: np.ndarray
+    b: np.ndarray
+
+    def __lt__(self, other):
+        # Consider regions with higher error estimates as being "less than" regions with
+        # lower order estimates, so that regions with high error estimates are placed at
+        # the top of the heap.
+        return _max_norm(self.error) > _max_norm(other.error)
+
+
+@dataclass
+class CubatureResult:
+    estimate: np.ndarray
+    error: np.ndarray
+    success: bool
+    status: str
+    regions: list[CubatureRegion]
+    subdivisions: int
+    atol: float
+    rtol: float
+
+
+def cub(f, a, b, rule="gk21", rtol=1e-05, atol=1e-08, max_subdivisions=10000,
+        args=(), kwargs=None):
+    r"""
+    Adaptive cubature of multidimensional array-valued function.
+
+    Given an arbitrary integration rule, this function returns an estimate of the
+    integral to the requested tolerance over the region defined by the arrays ``a`` and
```

Comment:
Here and elsewhere arguments should use single backticks
```suggestion
    integral to the requested tolerance over the region defined by the arrays `a` and
```